superset-frontend/src/dashboard/actions/hydrate.js

```diff
@@ -333,9 +348,21 @@ export const hydrateDashboard =
                 rootPath: [DASHBOARD_ROOT_ID],
                 excluded: [chartId], // By default it doesn't affects itself
               },
+              chartsInScope: Array.from(sliceIds),
             },
           };
         }
+        if (
+          behaviors.includes(Behavior.INTERACTIVE_CHART) &&
+          !metadata.chart_configuration[chartId].crossFilters?.chartsInScope
+        ) {
+          metadata.chart_configuration[chartId].crossFilters.chartsInScope =
```

Comment:
Optimization for cross filters (works the same as the one implemented for native filters a few months ago) to speed up finding charts in scope